### PR TITLE
[CI] Remove the GitHub macOS tests workflow

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR removes the GitHub macOS tests workflow.

### Motivation

The GitHub Actions macOS tests are the main source of [irregularly long MQ runs](https://app.datadoghq.com/ci/pipeline-executions?query=ci_level%3Apipeline%20%40git.branch%3Amq-working-branch-main-%2A%20%40ci.pipeline.name%3ADataDog%2Fdatadog-agent&agg_m=count&agg_m_source=base&agg_t=count&colorBy=meta%5B%27ci.stage.name%27%5D&colorByAttr=meta%5B%27ci.stage.name%27%5D&currentTab=trace&fromUser=false&graphType=flamegraph&index=cipipeline&spanViewType=overview&start=1733490763355&end=1734095563355&paused=false). Now that we have the job running and stable on the Gitlab macOS runners, we don't need the workflow anymore.

### CODE REVIEW

File: .gitlab/source_test/macos.yml [hunk 1]
```diff
@@ -2,39 +2,6 @@
 include:
   - .gitlab/common/macos.yml
 
-tests_macos:
-  stage: source_test
-  # HACK: Run macOS unit tests only on full pipelines, to limit the use of macOS GitHub runners.
-  rules:
-    - !reference [.on_main_or_release_branch_or_all_builds_or_pkg_installer_changes]
```
Comment (KSerrania) [inline .gitlab/source_test/macos.yml:9]: I _think_ this is the only place where this rule was used. Could you check and clean up?

File: .gitlab/source_test/macos.yml [hunk 2]
```diff
@@ -2,39 +2,6 @@
...
-  artifacts:
-    expire_in: 2 weeks
-    when: always
-    paths:
-      - test_output.json
-      - junit-*-repacked.tgz
-    reports:
-      junit: "**/junit-out-*.xml"
-
 .tests_macos_gitlab:
```
Comment (KSerrania) [inline .gitlab/source_test/macos.yml:5]: Should the Gitlab macOS tests run in the MQ instead? With this change, we go from a state where unit tests on macOS run sometimes in the MQ, to a state where they never run in the MQ.

File: .gitlab/source_test/macos.yml (hunk 1 above)
Comment (amenasria) [reply .gitlab/source_test/macos.yml:9]: Good point, fixed with [6c93173](https://github.com/DataDog/datadog-agent/pull/32151/commits/6c931736c37abec27954eb22912e7aab8075120f) !

File: .gitlab/source_test/macos.yml (hunk 2 above)
Comment (amenasria) [reply .gitlab/source_test/macos.yml:5]: Right ! Fixed with [8d5ddc3](https://github.com/DataDog/datadog-agent/pull/32151/commits/8d5ddc31b1cd58ae0a8131f7b744fa2be1f613ab)!

**Review by chouetz — APPROVED** [review]

**Pull request merged**